### DISCUSSION TITLE
Pip needs manual update on WSL Ubuntu 22.04.2

### DISCUSSION THREAD

**Discussion by crankbird** [imartinez/privateGPT#949] [General]
PSA - 
Just a quick FYI, in case anyone is trying to install using the pip3 install -r requirements.txt method on WSL running Ubuntu 22.04 and runs into an error that ends with 

"assert len(weights) == expected_node_count"

it seems that a fresh install of pip via 

sudo apt-get update
sudo apt install python3-pip

installs a slightly broken version of pip  (22.0.2)

to get around this try running

python3 -m pip install --upgrade pip

which installed (for me) 23.1.1, after which the requirements install seems to work just fine

**Comment by PasunuriSrinidhi** [comment]
@crankbird python3 -m pip install --upgrade pip ,after running this command and installed requirements but still got the same error . Can you please help me?   Thank you  

ERROR: Could not find a version that satisfies the requirement gpt4all==1.0.8 (from versions: 0.1.5, 0.1.6, 0.1.7)
ERROR: No matching distribution found for gpt4all==1.0.8
WARNING: There was an error checking the latest version of pip.
